### PR TITLE
Upgrade Stylo to 2024-01-16

### PR DESCRIPTION
**Pull request by Loirooriol**
This continues #32032

Changelog:
 - Upstream: https://github.com/servo/stylo/compare/bf13fe5...ca91506
 - Servo fixups: https://github.com/servo/stylo/compare/f642181...e801dd1

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] There are tests for these changes

### CODE REVIEW

File: components/layout/query.rs [hunk 1]
```diff
@@ -766,10 +766,14 @@ pub fn process_resolved_style_request<'dom>(
...
-        // Firefox returns blank strings for the computed value of shorthands,
-        // so this should be web-compatible.
-        PropertyId::ShorthandAlias(..) | PropertyId::Shorthand(_) => return String::new(),
+        PropertyId::NonCustom(id) => {
+            match id.unaliased().as_longhand() {
+                Some(id) => id,
+                // Firefox returns blank strings for the computed value of shorthands,
+                // so this should be web-compatible.
+                None => return String::new(),
+            }
```
Comment (mrobinson) [inline components/layout/query.rs:775]: Could this be simply: `id.unaliased().as_longhand().unwrap_or_default()`?

File: components/layout/query.rs [hunk 2]
```diff
@@ -811,10 +815,14 @@ fn process_resolved_style_request_internal<'dom>(
...
-        PropertyId::LonghandAlias(id, _) | PropertyId::Longhand(id) => id,
-        // Firefox returns blank strings for the computed value of shorthands,
-        // so this should be web-compatible.
-        PropertyId::ShorthandAlias(..) | PropertyId::Shorthand(_) => return String::new(),
+        PropertyId::NonCustom(id) => {
+            match id.unaliased().as_longhand() {
+                Some(id) => id,
+                // Firefox returns blank strings for the computed value of shorthands,
+                // so this should be web-compatible.
+                None => return String::new(),
```
Comment (mrobinson) [inline components/layout/query.rs:823]: Same comment here...

File: components/layout_2020/query.rs [hunk 3]
```diff
@@ -263,9 +263,9 @@ pub fn process_resolved_style_request_for_unstyled_node<'dom>(
     );
     let style = styles.primary();
     let longhand_id = match *property {
-        PropertyId::LonghandAlias(id, _) | PropertyId::Longhand(id) => id,
-        PropertyId::ShorthandAlias(id, _) | PropertyId::Shorthand(id) => {
-            return shorthand_to_css_string(id, style);
+        PropertyId::NonCustom(id) => match id.longhand_or_shorthand() {
+            Ok(lh) => lh,
+            Err(sh) => return shorthand_to_css_string(sh, style),
```
Comment (mrobinson) [inline components/layout_2020/query.rs:268]: I would go with `longhand` and `shorthand` here instead of `lh` and `sh`.

**Review by mrobinson — CHANGES_REQUESTED** [review]

**Review by mrobinson — APPROVED** [review]
Looks good modulo a couple nits above.

File: components/layout/query.rs (hunk 1 above)
Comment (Loirooriol) [reply components/layout/query.rs:775]: But there is no default LonghandId, the early return is needed.

File: components/layout_2020/query.rs (hunk 3 above)
Comment (Loirooriol) [reply components/layout_2020/query.rs]: Changed to `longhand_id` and `shorthand_id`.

File: components/layout/query.rs (hunk 1 above)
Comment (mrobinson) [reply components/layout/query.rs:775]: Ah, that makes sense!

**Pull request merged**